codice/ddf
# DDF-2279: Update Stub Server to support Range Headers

#### What does this PR do?
The stub server used for itests will now properly advertise that it supports byte range headers, and will respond to requests with range headers appropriately. All previous functionality is still available, including setting up a server without range header support.
#### Who is reviewing it 
@Lambeaux 
@cmthom10 
@oconnormi 
@bantillo 

#### Choose 2 committers to review/merge the PR 
@figliold 
@lessarderic

#### How should this be tested?
The best way to test at the moment is to run the Ignored test with `DEBUG` logging and ensure that the response's headers and status code are being set correctly based on the request's headers. The debug logging statements added in this PR should be enough to verify the headers are correct.

#### Any background context you want to provide?
While working this ticket, we discovered that range header support for DDF was lost somehow during the migration on DDF-Catalog. The itest added is currently set to `@Ignore` until range header support is fixed. There is currently a bug ticket out for it, and it will be worked as soon as possible.

Commit Messages:
 - Updated the restito stub server so it appropriately advertises
 that it can support a byte range-header.
 - Stub server will only respond with the specified range of bytes
 and will set the correct response status and headers when a
 range-header request is made.
 - Unfortunately, a bug was discovered and DDF does not currently
 handle partial content responses when a range header request is
 made, resulting in corrupted data. These tests are @Ignore until
 that bug ticket is completed.
